txpipe/lssweights.py

```diff
@@ -0,0 +1,1103 @@
+		sys_meta = {'masked':True,'normed':False}
+		return sys_maps, sys_names, sys_meta
+
+	def get_deltag(self, tomobin):
+		"""
+		convert the ra, dec of the lens sample to pixel number counts
+		"""
+		import healpy as hp
+		import numpy as np  
+		import healsparse as hsp
+
+		with self.open_input("mask", wrapper=True) as map_file:
+			nside = map_file.read_map_info("mask")["nside"]
+			nest = map_file.read_map_info("mask")["nest"]
+			mask = map_file.read_map("mask")
+
+		#load the ra and dec of this lens bins
+		with self.open_input("binned_lens_catalog_unweighted", wrapper=False) as f:
+			ra = f[f"lens/bin_{tomobin}/ra"][:]
+			dec = f[f"lens/bin_{tomobin}/dec"][:]
+			weight = f[f"lens/bin_{tomobin}/weight"][:]
+
+		#pixel ID for each lens galaxy
+		obj_pix = hp.ang2pix(nside,ra,dec,lonlat=True, nest=True)
+
+		Ncounts_bincount = np.bincount(obj_pix, weights=weight)
+		pixels_bincount = np.arange(len(Ncounts_bincount))
+		
+		pixel = pixels_bincount[Ncounts_bincount != 0.]
+		Ncounts = Ncounts_bincount[Ncounts_bincount != 0.]
+
+		if nest:
+			maskpix = np.where(mask!=hp.UNSEEN)[0]
+		else:
+			maskpix = hp.ring2nest(nside, np.where(mask!=hp.UNSEEN)[0])
+
+		#fractional coverage
+		frac = hsp.HealSparseMap.make_empty(32, nside, dtype=np.float64)
+		frac.update_values_pix(maskpix, mask[np.where(mask!=hp.UNSEEN)[0]])
+
+		deltag = hsp.HealSparseMap.make_empty(32, nside, dtype=np.float64)
+		deltag.update_values_pix(maskpix, 0.0)
+		deltag.update_values_pix( pixel, Ncounts )
+		n = deltag[maskpix]/frac[maskpix]
+		nmean = np.average(n, weights=frac[maskpix])
+		deltag.update_values_pix( maskpix, n/nmean-1.0 ) #overdenity map 
+
+		return deltag, frac
+
+
+	def load_and_mask_sysmaps(self):
+		"""
+		load the SP maps and mask them
+		"""
+		import numpy as np 
+		import healpy as hp
+		import healsparse as hsp
+
+		s = time.time()
+
+		root = self.config["supreme_path_root"]
+		sys_files = glob.glob(f"{root}*.hs")
+		nsys = len(sys_files)
+		print(f"Found {nsys} total systematic maps")
+
+		with self.open_input("mask", wrapper=True) as map_file:
+			mask = map_file.read_map("mask")
+			mask_nest = map_file.read_map_info("mask")["nest"]
+			mask = hsp.HealSparseMap(
+				nside_coverage=32, 
+				healpix_map=(mask==hp.UNSEEN).astype('int'), 
+				nest=mask_nest, sentinel=0)
+			nside = map_file.read_map_info("mask")["nside"]
+
+		sys_maps = []
+		sys_names = []
+		for i, map_path in enumerate(sys_files):
+			# strip root, .hs, and underscores to get friendly name
+			sys_name = map_path[len(root) : -3].strip("_")
+			sys_names.append(sys_name)
+
+			# get actual data for this map
+			sys_map = self.read_healsparse(map_path, nside)
+
+			sys_map.apply_mask(mask, mask_bits=1)
+
+			#apply mask
+			sys_maps.append(sys_map)
+
+		f = time.time()
+		print("load_and_mask_sp_maps took {0}s".format(f-s))
+
+		return np.array(sys_maps), np.array(sys_names)
+
+	def calc_1d_density(self, tomobin, Fmap=None):
```

Comment:
Could you give Fmap a more descriptive name?

Reply:
changed to mean_density_map

Reply:
Have you pushed this change? It's still called Fmap on github.

Reply:
yes, sorry i hadn't pushed a few changes, should be there now